src/components/app/index.tsx

```diff
@@ -23,13 +25,31 @@ export default class App extends Component<Props, State> {
     }
   }
 
+  private async getImageData(bitmap: ImageBitmap): Promise<ImageData> {
+    // Make canvas same size as image
+    const canvas = document.createElement('canvas');
+    [canvas.width, canvas.height] = [bitmap.width, bitmap.height];
+    // Draw image onto canvas
+    const ctx = canvas.getContext('2d');
+    if (!ctx) {
+      throw new Error("Could not create canvas contex");
+    }
+    ctx.drawImage(bitmap, 0, 0);
+    return ctx.getImageData(0, 0, bitmap.width, bitmap.height);
+  }
+
   @bind
   async onFileChange(event: Event) {
     const fileInput = event.target as HTMLInputElement;
     if (!fileInput.files || !fileInput.files[0]) return;
     // TODO: handle decode error
-    const img = await createImageBitmap(fileInput.files[0]);
-    this.setState({ img });
+    const bitmap = await createImageBitmap(fileInput.files[0]);
+    const data = await this.getImageData(bitmap);
+    const encoder = new MozJpegEncoder();
```

Comment:
The way it’s currently written, we should just be able to load these encoders in a worker and invoke `encode` via Comlink.